pkg/datagatherer/k8s/client.go

```diff
@@ -4,21 +4,39 @@ package k8s
 import (
 	"github.com/pkg/errors"
 	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 )
 
 // NewDynamicClient creates a new 'dynamic' clientset using the provided kubeconfig.
 // If kubeconfigPath is not set/empty, it will attempt to load the InClusterConfig.
 func NewDynamicClient(kubeconfigPath string) (dynamic.Interface, error) {
-	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
-	if err != nil {
-		return nil, errors.WithStack(err)
+	var cfg *rest.Config
+	var err error
+	switch kubeconfigPath {
+	// If the kubeconfig path is not provided, use the default loading rules
+	// so we read the regular KUBECONFIG variable.
+	case "":
+		apicfg, err := clientcmd.NewDefaultClientConfigLoadingRules().Load()
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+		cfg, err = clientcmd.NewDefaultClientConfig(*apicfg, &clientcmd.ConfigOverrides{}).ClientConfig()
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+	// Otherwise use the explicitly named kubeconfig file.
+	default:
+		cfg, err = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+			&clientcmd.ClientConfigLoadingRules{ExplicitPath: kubeconfigPath},
+			&clientcmd.ConfigOverrides{}).ClientConfig()
+		if err != nil {
+			return nil, errors.WithStack(err)
```

Comment:
I'm not too sure if we _really_ want `errors.WithStack(err)` all over the place, and feel like it may be more sensible to only print this stack trace as and when it may be useful. But I didn't want to break convention here 😄